### ISSUE TITLE
webgpu integration

### ISSUE THREAD

**Issue by mreinstein** [kmamal/node-sdl#20]
Would be amazing to be able to leverage webgpu with sdl, similar to how there are canvas and gl examples.

I'm wondering if this module might make it possible?  https://www.npmjs.com/package/webgpu

**Comment by kmamal** [comment]
I didn't realize there were libraries for WebGPU already. I'll have to look into it. I'm planning to support this, but it could take a while.

**Comment by mreinstein** [comment]
That said this library I linked to, I'm just noticing it hasn't been updated in 3 years. I wonder if node-sdl could instead link directly against dawn, and a more recent version.

**Comment by kmamal** [comment]
@mreinstein Here's a start: https://www.npmjs.com/package/@kmamal/gpu. I was able to build and package the dawn_node bindings. It's still a "headless" solution, meaning you can't render _directly_ to the window, but you can still render to a buffer, and display the result with `window.render()` as in the second example.

Now I just need to figure out how to get a webgpu texture from a window...

**Comment by mreinstein** [comment]
wow! This is amazing progress!!

**Comment by mreinstein** [comment]
@kmamal I could sponsor work on this if it's something you'd be interested in furthering.  I'm currently using electron for an upcoming game but I'd love to try to leverage node-sdl + kmamal/gpu instead for a leaner build.

**Comment by kmamal** [comment]
@mreinstein I've sent you an email

**Comment by kmamal** [comment]
@mreinstein I have released a new version that supports rendering directly to a window. I could use some help testing this as I have limited access to Mac and Windows machines. I know it works on Linux/X11/Vulkan.

**Comment by mreinstein** [comment]
amazing! I should have some bandwidth next week to evaluate this. Looking forward to it!

**Comment by mreinstein** [comment]
I have both windows and mac, can def evaluate those.

**Comment by kmamal** [comment]
@mreinstein released new versions of `@kmamal/sdl` and `@kmamal/gpu`. It should run without crashing on MacOS and Windows now. I'd be grateful if you could test the readme example again whenever you have some time.

**Comment by mreinstein** [comment]
Nice!  I'll check it out tonight after the kid goes to sleep.  Looking forward to testing this!

**Comment by kmamal** [comment]
Does the red corner button work? Not too familiar with MacOS. The cmd+Q shortcut might be something each application has to register itself rather then being a sytem-wide shortcut like alt+f4 on windows.

**Comment by mreinstein** [comment]
Yes, the red corner button _does_ work. `cmd + q` along with the menubar quit function don't work (this one):

<img width="360" alt="Screenshot 2023-11-29 at 8 31 00 PM" src="https://github.com/kmamal/node-sdl/assets/718067/6cb445b1-a5bf-4f60-b791-69937451aac2">

**Comment by mreinstein** [comment]
Windows works too!

**Comment by mreinstein** [comment]
This is amazing, I'm super excited to start playing with this!
